gce_workload_certs_refresh/main.go

```diff
@@ -0,0 +1,290 @@
+//  Copyright 2022 Google LLC
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+// GoogleAuthorizedKeys obtains SSH keys from metadata.
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/GoogleCloudPlatform/guest-logging-go/logger"
+)
+
+const (
+	contentDirPrefix  = "/run/secrets/workload-spiffe-contents"
+	tempSymlinkPrefix = "/run/secrets/workload-spiffe-symlink"
+	symlink           = "/run/secrets/workload-spiffe-credentials"
+)
+
+var (
+	programName    = "gce_workload_certs_refresh"
```

Comment:
Can't we use os.Args[0] instead of hardcoding it?

Reply:
it is a repo norm to have this path, the reason is due to multi-OS support. in this particular binary it doesn't matter but i was just defaulting to following the norm. is there any practical benefit, do you think?

Reply:
Not having to repeat hardcoded and boilerplate maybe(?) if we have this norm then let the code follow the norm and not us, abstract it, make it reasonable to be used across the code base and reduce the boilerplate/hardcoded. It seems harmless in this case - it's just a mindset, this whole logging stuff could be abstracted keeping it all consistent across the code base. I wouldn't block this PR because of that though :-)

Reply:
i totally agree, we went from 2 binaries to 4 this year so it is probably worth creating an extracted 'envelope' that does some of these now-common things. just didn't want to link that to this PR

Reply:
ok